docs/design/datacontracts/ComWrappers.md

```diff
@@ -15,19 +23,81 @@ Data descriptors used:
 | Data Descriptor Name | Field | Meaning |
 | --- | --- | --- |
 | `NativeObjectWrapperObject` | `ExternalComObject` | Address of the external COM object |
+| `ManagedObjectWrapperHolderObject` | `WrappedObject` | Address of the wrapped object |
+| `ManagedObjectWrapperLayout` | `RefCount` | Reference count of the managed object wrapper |
+| `ComWrappersVtablePtrs` | `MowQueryInterface` | Function pointer of ManagedObjectWrapper_QueryInterface |
+| `ComWrappersVtablePtrs` | `TtQueryInterface` | Function pointer of TrackerTarget_QueryInterface |
 
 Global variables used:
 | Global Name | Type | Purpose |
 | --- | --- | --- |
+| `ComWrappersVtablePtrs` | TargetPointer | Pointer to struct containing ComWrappers-related function pointers |
+| `DispatchThisPtrMask` | TargetPointer | Used to mask low bits of CCW pointer to the nearest valid address from which to read a managed object wrapper |
+| `NativeObjectWrapperClass` | ushort | The index of the native object wrapper class into the CoreLibBinder classes |
```

Comment:
Can we avoid exposing the CoreLibBinder and this index as part of the data contract, and instead make the type name part of the contract? I think it is ok to hardcode the type name in the debugger/cDAC and note it as a comment next to the type name in the sources by comment like "// debugger depends on this name". 

It is the current way to do this that's used in number of places in CoreLib, one example from many https://github.com/dotnet/runtime/blob/8c362f7253a6e07ce02a3fe01babe2c677e5bb45/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncMethodBuilderCore.cs#L15